runtime/runtime/src/lib.rs

```diff
@@ -1294,62 +1297,125 @@ impl Runtime {
             })?;
         Ok(())
     }
+}
+
+struct StorageComputer<'a> {
+    result: HashMap<String, u64>,
+    config: &'a StorageUsageConfig,
+}
```

Comment:
Do we need this doer object? At least in the current code, seems like it can be just a function? Even when we add straming here, I feel we might get by with a function?

```rust
    /// It's okay to use unsafe math here, because this method should only be called on the trusted
    /// state records (e.g. at launch from genesis)
    pub fn compute_storage_usage(
        &self,
        genessis: &Genesis,
        config: &RuntimeConfig,
    ) -> HashMap<AccountId, u64> {
        let mut res = HashMap::new();
        genessis.process_records(|record| {
            ...
        });
        res
    }
```

More generally, a lot of code in this PR adds objects which shoudl be used according to a `create, process*, finalize` protocol. I think some of that is inherent to the problem, but it'd be cool to minimize this where possible.

Reply:
Ah, we *do* need this for `GenesisStateApplier`... :thinking: yeah, don't have any suggestions on how to make this better...

Reply:
Yeah, for now StorageComputer stays